### PR TITLE
feat: Add an optional Flight client to the IOx client library

### LINKED ISSUE #731: Add optional Flight client support to influxdb_iox_client
Client side of https://github.com/influxdata/influxdb_iox/pull/712, using a Cargo feature so that the `arrow` deps are only included if requested.

### PR DESCRIPTION
**Pull request by carols10cents**
Fixes #731.

### CODE REVIEW

File: influxdb_iox_client/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```toml
+[dependencies]
+influxdb_iox_client = { version = "[..]", features = ["flight"] }
+```
+
+Then you can create an `influxdb_iox_client::FlightClient` by using the `FlightClientBuilder`. Here
+is an example of creating an instance that connects to an IOx server with its gRPC services
+available at `http://localhost:8082` and using it to perform a query:
+
+```rust
```
Comment (alamb) [inline influxdb_iox_client/README.md:42]: 👍

File: influxdb_iox_client/src/client/flight.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+/// # async fn test() {
+/// use data_types::database_rules::DatabaseRules;
+/// use influxdb_iox_client::FlightClientBuilder;
+///
+/// let client = FlightClientBuilder::default()
+///     .build("http://127.0.0.1:8082")
+///     .expect("client should be valid");
+///
+/// let mut query_results = client
+///     .perform_query(scenario.database_name(), sql_query)
```
Comment (alamb) [inline influxdb_iox_client/src/client/flight.rs:33]: I am surprised this passes the `--doc` tests as I don't see where `sql_query` is defined

File: tests/end_to_end_cases/flight_api.rs [hunk 3]
```diff
@@ -1,69 +1,21 @@
...
-use std::{convert::TryFrom, sync::Arc};
-
-// TODO: this should be shared
-#[derive(Serialize, Debug)]
-struct ReadInfo {
-    database_name: String,
-    sql_query: String,
-}
+use arrow_deps::assert_table_eq;
+use influxdb_iox_client::FlightClientBuilder;
```
Comment (alamb) [inline tests/end_to_end_cases/flight_api.rs:3]: 👍

**Review by alamb — APPROVED** [review]
Very cool -- I have nothing of substance to add. I vote :shipit:

**Review by domodwyer — APPROVED** [review]
It's so cooooool 🚀

File: influxdb_iox_client/src/client/flight.rs (hunk 2 above)
Comment (carols10cents) [reply influxdb_iox_client/src/client/flight.rs:33]: o.O that *is* weird... investigating...

Comment (carols10cents) [reply influxdb_iox_client/src/client/flight.rs]: Ok, so if your doc test doesn't include `fn main`, then your code gets wrapped in `fn main`, and [doesn't get run as a test](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=002aae128923f9d57cfa4274321b478b) as intended here.

**Pull request merged**